packages/binaryen/binaryen.0.11.0/opam

```diff
@@ -6,7 +6,7 @@ license: " Apache-2.0"
 homepage: "https://github.com/grain-lang/binaryen.ml"
 bug-reports: "https://github.com/grain-lang/binaryen.ml/issues"
 depends: [
-  "ocaml" {>= "4.12"}
+  "ocaml" {>= "4.12" < "4.13"}
```

Comment:
If it’s windows specific what about this?
```suggestion
  "ocaml" {>= "4.12"}
  "ocaml" {< "4.13" & os = "win32"}
```

Reply:
I actually don't think esy supports that syntax 🤔 Do you know if it does?

Reply:
it adds the `< 4.13` constraint if the current OS is windows

Reply:
Esy doesn't apply the OS to the constraint system. I had a PR open for it, but it can't be accepted because those can't be written into the lockfile. https://github.com/esy/esy/pull/1349

Reply:
From opam's perspective, this package doesn't work properly on Windows/4.12 either as far as I can tell - it's only working at all in setup-ocaml because it's pulling in tools from unrelated distros? I'd confirm that 4.12 works on a stable set-up first on Windows before digging into the 4.12/4.13 differences.

Reply:
That's not really useful help. We definitely work with esy on Windows.

Reply:
Where are you pulling cmake and python 3 from?